### DISCUSSION TITLE
Saving and loading nested state to JSON and from JSON (store serialization).

### DISCUSSION THREAD

**Discussion by bukowa** [vuejs/pinia#2688] [Help and Questions]
Hey I am looking for a simple way to save and load my store to `JSON`.
I wonder what's the best way for this serialization to work out of the box?
I wonder if there is any standard for this or / my approach is wrong?
```js
let store = useWorkspaceCollectionStore();
let store2 = useWorkspaceStore()
store.workspaces["test"] = store2;
store.$state = JSON.parse(JSON.stringify(store.$state));
```

**Comment by kevinramharak** [comment]
Maybe take a look at [pinia-plugin-persistedstate](https://github.com/prazdevs/pinia-plugin-persistedstate). There are other plugins that provide similar functionality, if you still want to roll your own, you can at least take a look at their implementations and get some inspiration. Serializing to/from JSON should work fine for most cases.
